src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +62,49 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (isProperTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
+        setGamePrefab();
+        int oppositeTeamCount = 0;
+        int teleporterTeamCount = 0;
+        int maxTeamSizeDifference = gameEntity.getComponent(LASConfigComponent.class).maxTeamSizeDifference;
+        String teleporterTeam = teleporter.getComponent(LASTeamComponent.class).team;
+        Iterable<EntityRef> characters = entityManager.getEntitiesWith(PlayerCharacterComponent.class,
+                LASTeamComponent.class);
+
+        for (EntityRef character : characters) {
+            String otherPlayerTeam = character.getComponent(LASTeamComponent.class).team;
+            if (teleporterTeam.equals(otherPlayerTeam)) {
+                teleporterTeamCount++;
+            } else if (!otherPlayerTeam.equals(LASUtils.WHITE_TEAM)) {
+                oppositeTeamCount++;
+            }
```

Comment:
Afaik there's currently no possibility to join the white team (and the white team is actually on of the aspects of the gameplay that's targeting at the original gameplay idea, but is not relevant to the current state of LightAndShadow. In the context of cutting things down, I'd say, we should rather remove it from LaS instead of further integrating it in logic we clean up or build

Reply:
Yeah currently the white team only helps identify the lobby players, so if that's not needed should I remove all instances of it?

Reply:
Yes I'd say so

Reply:
Where did you find that?

Reply:
![image](https://user-images.githubusercontent.com/52852877/117751403-ad44ac80-b232-11eb-8a16-f29d3c96b6ed.png)
![image](https://user-images.githubusercontent.com/52852877/117751498-d82f0080-b232-11eb-9b01-109bb625418f.png)
Multiplayer is glitching a bit for me right now, have to clean my workspace up before I can confirm but in ```ClientSkinSystem``` , ```onCreateDefaultVisualCharacter``` initializes the ```skeletalMeshComponent``` based on the initial team of the player and ```onAwaitedLocalCharacterSpawnEvent``` uses ```setHealthHUD``` to initialize the healthBar based on the initial team of the player, so lobby players should look different and have different health bars.